### PR TITLE
Allow relatedTo for fields

### PR DESCRIPTION
**Pull request by timkelty**
### Description
Adds an autosuggest input for entering `relatedTo` IDs in a field context.

Did some rough adjustment for display, but might need to think about guidelines for other rules that might be multi-line. For now I just aligned the row to top and put the label in placeholder.

We also talked about having some kind of warning about the danger of putting in IDs directly.

CleanShot 2022-02-13 at 22 55 29@2x

### CODE REVIEW

**Comment by brandonkelly** [comment]
Not working locally for me:

Screenshot of a relation condition rule, where the input is not visible

Screenshot of JavaScript error output

**Comment by timkelty** [comment]
@brandonkelly should be fixed now.

**Comment by timkelty** [comment]
@brandonkelly This is technically working now, but I'm switching it to draft until @lukeholder has a chance to ask the HTMX team about what exactly is going on and what event we should be using.

**Comment by timkelty** [comment]
…or not. Apparently I can't switch it to back to a draft. At any rate – don't merge yet :)

**Comment by brandonkelly** [comment]
Converted the PR to a draft for the time being.

**Comment by timkelty** [comment]
After some back and forth with @lukeholder and the HTMX folks, the best way to inject our JS dependencies prior to `htmx:load` is to use the `transformResponse` method.

The are considering adding a `htmx:newContent` event, which when implemented, we would be able to use for a cleaner approach.

**Comment by brandonkelly** [comment]
Merged!

File: src/web/assets/htmx/src/htmx.js [hunk 1]
```diff
@@ -9,19 +9,22 @@ htmx.defineExtension('craft-cp', {
...
             return;
         }
 
-        const allHeadHtml = evt.detail.elt.querySelectorAll("template.hx-head-html");
-        const allBodyHtml = evt.detail.elt.querySelectorAll("template.hx-body-html");
+        const allHeadHtml = doc.querySelectorAll("template.hx-head-html");
+        const allBodyHtml = doc.querySelectorAll("template.hx-body-html");
 
         for (let i = 0; i < allHeadHtml.length; i++) {
             const headHtml = allHeadHtml[i].innerHTML;
```
Comment (lukeholder) [inline src/web/assets/htmx/src/htmx.js:30]: Maybe we should be using https://developer.mozilla.org/en-US/docs/Web/API/Node/textContent

as: 

> textContent gets the content of all elements, including [<script>](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script) and [<style>](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/style) elements. In contrast, innerText only shows "human-readable" elements.

File: src/web/assets/htmx/src/htmx.js [hunk 2]
```diff
@@ -37,8 +40,11 @@ htmx.defineExtension('craft-cp', {
             }
         }
 
+        return text;
```
Comment (lukeholder) [inline src/web/assets/htmx/src/htmx.js:43]: Maybe we should remove the template tags from the `doc` node and return `doc.textContent` here?

**Review by lukeholder — COMMENTED** [review]
Can discuss on slack.

File: src/web/assets/htmx/src/htmx.js (hunk 1 above)
Comment (lukeholder) [reply src/web/assets/htmx/src/htmx.js:30]: Sorry just realized `innerHTML` was being used, not innerText. Ignore.

**Pull request merged**